modules/ppcp-applepay/resources/js/ApplepayButton.js

```diff
@@ -160,26 +163,68 @@ console.log('[ApplePayButton] wrapper', wrapper)
     /**
      * Show Apple Pay payment sheet when Apple Pay payment button is clicked
      */
-    onButtonClick() {
+    async onButtonClick() {
         const paymentDataRequest = this.paymentDataRequest();
-        console.log('[ApplePayButton] onButtonClick: paymentDataRequest', paymentDataRequest, this.context);
-
+        // trigger woocommerce validation if we are in the checkout page
+        if (this.context === 'checkout') {
+            const checkoutFormSelector = 'form.woocommerce-checkout';
+            const errorHandler = new ErrorHandler(
```

Comment:
I think creating multiple ErrorHandler objects (here and old `button.js`) can sometimes cause some issues with displayed messages.

But not sure about the best way to fix it.